src/components/TabStepper/components/TabStep/TabStep.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useMemo } from 'react';
+import { Tab as ReactAriaTab } from 'react-aria-components';
+
+import { stepContainer, stepStyles, stepSubtitle, stepTitle } from './TabStep.style';
+import type { TabStepProps } from '../../types';
+
+import Icon from '~/components/Icon';
+import useTheme from '~/hooks/useTheme';
+
+const TabStep = React.forwardRef<HTMLDivElement, TabStepProps>((props, ref) => {
+  const { children, id, title, subtitle, status = 'pending', dataTestPrefixId, ...rest } = props;
+
+  const hasIcon = status !== 'pending';
+
+  const theme = useTheme();
+
+  const iconName = useMemo(() => {
```

Comment:
Nit: Here we could simplify  the `iconName` and `iconColor` for brevity and remove `useMemo`. I am fine thought keeping it as is. You can decide if it's worth it. e.g.
```
const STATUS_ICON = {
  done: { name: 'success', color: 'textColor.default.active' },
  error: { name: 'warning', color: 'indicators.error' }
};

const icon = status !== 'pending' ? STATUS_ICON[status] : null;

{icon && (
    <Icon
      name={icon.name}
      color={theme.tokens.colors.get(icon.color)}
      dataTestPrefixId={`${dataTestPrefixId}_icon`}
    />
 )}
```